edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs

```diff
@@ -209,7 +209,11 @@ void RegisterAmqpModule(ContainerBuilder builder)
             TimeSpan configUpdateFrequency = TimeSpan.FromSeconds(configUpdateFrequencySecs);
             bool checkEntireQueueOnCleanup = this.configuration.GetValue("CheckEntireQueueOnCleanup", false);
             bool closeCloudConnectionOnDeviceDisconnect = this.configuration.GetValue("CloseCloudConnectionOnDeviceDisconnect", true);
-            bool isLegacyUpstream = this.configuration.GetValue("mqttBrokerSettings:legacyUpstream", true);
+
+            bool isLegacyUpstream = !experimentalFeatures.Enabled
```

Comment:
>experimentalFeatures.Enabled [](start = 37, length = 28)

this is not needed, both experimentalFeatures.Enabled is the master switch and also taken into account when calculating EnableMqttBroker and EnabledNestedEdge